### DISCUSSION TITLE
Bad input argument using simple function

### DISCUSSION THREAD

**Discussion by adrianhelmlingcornell** [aesara-devs/aesara#1000] [Q&A]
## Description of your problem or feature request

I get the following error when trying to pretty-print my first aesara functions:

**Please provide a minimal, self-contained, and reproducible example.**
```python
import aesara
import aesara.tensor as at
import numpy as np
from scipy.special import gamma

c1 = 1
c2 = gamma(5)  # Euler Gamma function
z = at.dscalar('z')

f1 = aesara.function([z], c1 * z)
f2 = aesara.function([z], c2 * z)

print(aesara.pp(f1(z)), aesara.pp(f2(z)))
```

**Please provide the full traceback of any errors.**
```python
  File "PATH_TO_SCRIPT", line 30, in <module>
    print(aesara.pp(varphi_t_func(z)), aesara.pp(varphi_r_func(z)))
  File "/home/adrian/anaconda3/envs/py39/lib/python3.10/site-packages/aesara/compile/function/types.py", line 863, in __call__
    s.storage[0] = s.type.filter(
  File "/home/adrian/anaconda3/envs/py39/lib/python3.10/site-packages/aesara/tensor/type.py", line 134, in filter
    raise TypeError(
TypeError: Bad input argument with name "z" to aesara function with name "PATH_TO_SCRIPT:27" at index 0 (0-based).  
Backtrace when that variable is created:

trace 0
  File "PATH_TO_SCRIPT", line 22, in <module>
    z = at.dscalar('z')
trace 1
  File "PATH_TO_SCRIPT", line 24, in <module>
    f1 = c1 * z  # actual combination of numeric variables
Expected an array-like object, but found a Variable: maybe you are trying to call a function on a (possibly shared) variable instead of a numeric array?
```

**Please provide any additional information below.**

## Versions and main components

* Aesara version: 2.7.2
* Aesara config: See here: [aesara_config.txt](https://github.com/aesara-devs/aesara/files/8915874/aesara_config.txt)
* Python version: 3.10.5
* Operating system: Ubuntu 18l.04
* How did you install Aesara: conda

**Comment by ricardoV94** [comment]
When you call the function you have to pass a value to it, not the original variable `z`. So something like `f1(2.0)`. This is exactly what the error message suggests. 

Also not sure what you are trying to do with the `aesara.pp` call. It should be `aesara.pp(f1)`, otherwise you would be trying to pretty print the output of the function which would be just a number.
